impl/src/fmt/display.rs

```diff
@@ -94,8 +96,14 @@ fn expand_struct(
             .ident
             .clone()
             .map_or_else(|| syn::Member::Unnamed(i.into()), syn::Member::Named);
-        quote! {
-            let #var = &self.#member;
+        if let syn::Type::Reference(_) = f.ty {
```

Comment:
Afaict this only works for references, and will fail as soon as you use other pointer types, such as `Box<T>`.

Reply:
You are right, I will think about it more.